### ISSUE TITLE
Shared imports between Runestone Components and the Runestone server

### ISSUE THREAD

**Issue by bjones1** [RunestoneInteractive/RunestoneComponents#882]
This continues the discussion in #876 as a separate issue.

Goal: A way to provide common imports shared between Runestone Components and the Runestone server.

Constraints: Because these imports may be used in controllers or models of web2py, they must be minimal. That is, they must avoid imports of unrelated code (such as Runestone Components-only Sphinx `setup()` functions) or expensive operations (creating a database connection) which aren't required for the correct operation of the import.

Uses:
- Replace the `from runestone import cmap` and `from runestone.server.lp_common_lib import STUDENT_SOURCE_PATH, ...` statements in the Runestone server.
- Possibly: share relevant model files, so we can use DAL instead of re-creating models in SQLAlchemy to modify the database in Runestone Components.

Challenges:
- Currently, the Runestone server imports from Runestone Components. This isn't the best strategy, since it ends up importing a lot of components-only symbols.
- The Runestone Components `conf.py` file contains `from runestone import runestone_static_dirs, runestone_extensions`. Since this is now a part of each book, changing it would be painful. The `runestone.__init__` module begins with wildcard imports of all directives, making it a poor candidate for sharing with the server. In addition, it imports some command-line stuff (paver), which also isn't needed by the server.

Approaches:
- Create another package, perhaps called `runestone_common`, so we don't have to rethink/refactor `runestone`. Objection: two packages in one distribution seems a bit messy.
- One alternative: put all such imports in the web2py `modules` directory, which is evaluated only once. Then, we don't really care about performance. But a bit ugly from a modularity perspective.

**Comment by bnmnetp** [comment]
Another option may just be to clean up what we do at the top and intermediate levels of the runestone package.

I probably went overboard / didn't give it much thought when I added all of the lines in the top level __init__.py file where I do from .activecode import * . etc.  In fact this doesn't make a lot of sense at every component has a setup which is then imported.  What does it even mean to say `from runestone import setup` I'm assuming we get the last one.  I think that the top level __init__ file just wants to import the appropriate class so that it can use the docstring in cmap.

runestone_static_dirs and runestone_extensions and get_master_url are already defined in the __init__ file at the top level.

The only thing that is really forcing all of this importing is cmap.  Perhaps that should be pushed down a level or redesigned in a different way.
